content/kb/tutorials/databases/snowflake.md

```diff
@@ -107,39 +100,39 @@ Copy the code below to your Streamlit app and run it. Make sure to adapt the que
 import streamlit as st
 
 # Initialize connection.
-conn = st.experimental_connection('snowpark')
+conn = st.connection("snowflake")
 
 # Perform query.
-df = conn.query('SELECT * from mytable;', ttl=600)
+df = conn.query("SELECT * from mytable;", ttl=600)
 
 # Print results.
 for row in df.itertuples():
     st.write(f"{row.NAME} has a :{row.PET}:")
 ```
 
-See `st.experimental_connection` above? This handles secrets retrieval, setup, query caching and retries. By default, `query()` results are cached without expiring. In this case, we set `ttl=600` to ensure the query result is cached for no longer than 10 minutes. You can also set `ttl=0` to disable caching. Learn more in [Caching](/library/advanced-features/caching).
+See `st.connection` above? This handles secrets retrieval, setup, query caching and retries. By default, `query()` results are cached without expiring. In this case, we set `ttl=600` to ensure the query result is cached for no longer than 10 minutes. You can also set `ttl=0` to disable caching. Learn more in [Caching](/library/advanced-features/caching).
 
 If everything worked out (and you used the example table we created above), your app should look like this:
 
 ![Finished app screenshot](/images/databases/snowflake-app.png)
 
 ### Using a Snowpark Session
 
-The same [SnowparkConnection](/library/api-reference/connections/st.connections.snowparkconnection) used above also provides access to the [Snowpark Session](https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/session.html) for DataFrame-style operations that run natively inside Snowflake. Using this approach, you can rewrite the app above as follows:
+The same [SnowflakeConnection](/library/api-reference/connections/st.connections.snowflakeconnection) used above also provides access to the [Snowpark Session](https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/session.html) for DataFrame-style operations that run natively inside Snowflake. The [Snowpark Python library](https://docs.snowflake.com/en/developer-guide/snowpark/python/index) is a dependency for this, so be sure to `pip install snowflake-snowpark-python` first. Using this approach, you can rewrite the app above as follows:
 
 ```python
 # streamlit_app.py
 
 import streamlit as st
 
 # Initialize connection.
-conn = st.experimental_connection('snowpark')
+conn = st.connection("snowflake")
 
 # Load the table as a dataframe using the Snowpark Session.
 @st.cache_data
 def load_table():
-    with conn.safe_session() as session:
-        return session.table('mytable').to_pandas()
+    with conn.session() as session:
+        return session.table("mytable").to_pandas()
```

Comment:
Nit: we don't need the context manager anymore, this can just be

```
session = conn.session()
return session.table("mytable").to_pandas()
```
(Sorry I missed it on the first pass)